Sofa/Component/Constraint/Lagrangian/Correction/src/sofa/component/constraint/lagrangian/correction/LinearSolverConstraintCorrection.inl

```diff
@@ -73,19 +73,34 @@ void LinearSolverConstraintCorrection<DataTypes>::init()
     linearsolvers.clear();
 
     std::stringstream tmp ;
-    if (solverNames.size() == 0)
+    if (solverNames.empty())
     {
-        linearsolvers.push_back(c->get<sofa::core::behavior::LinearSolver>());
+        sofa::core::behavior::LinearSolver* s = nullptr;
+        c->get(s);
+        if(s)
+        {
+            if (s->getTemplateName() == "GraphScattered")
+                msg_warning() << "Can not use the solver " << s->getName() << " because it is templated on GraphScatteredType";
```

Comment:
I would like to know more about the reason GraphScattered types are not compatible with LinearSolverConstraintCorrection please.